api/provider/aws/common.go

```diff
@@ -7,8 +7,48 @@ import (
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"github.com/quintilesims/layer0/api/tag"
+	"github.com/quintilesims/layer0/common/errors"
 )
 
+func lookupEntityEnvironmentID(store tag.Store, entityType, entityID string) (string, error) {
+	tags, err := store.SelectByTypeAndID(entityType, entityID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("environment_id").First(); ok {
+		return tag.Value, nil
+	}
+
+	return "", errors.Newf(errors.EnvironmentDoesNotExist, "Could not resolve environment ID for %s '%s'", entityType, entityID)
+}
+
+func lookupTaskDefinitionFamily(store tag.Store, deployID string) (string, error) {
+	tags, err := store.SelectByTypeAndID("deploy", deployID)
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
will probably want another `if len(tags) == 0 { return errors.Newf(errors.DeployDoesNotExist, "") }`

Reply:
However, outside of that scenario, I'm not so confident in returning a `DeployDoesNotExist` error when we are missing only one particular tag.

Reply:
✅